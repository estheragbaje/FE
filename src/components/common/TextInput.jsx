import React from 'react';
import PropTypes from 'prop-types';

function TextInput({ id, onChange, value, name, required, title, type }) {
  return (
    <>
      <label htmlFor={id} className="d-flex">
        {title}
      </label>
      <input
        type={type}
        id={id}
        className="form-control mb-4"
        name={name}
        value={value}
        onChange={onChange}
        required={required}
      />
    </>
  );
}

TextInput.propTypes = {
  id: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  required: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default TextInput;
