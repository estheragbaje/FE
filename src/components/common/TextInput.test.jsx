import React from 'react';
import { shallow } from 'enzyme';

import TextInput from './TextInput';

describe('<TextInput />', () => {
  it('renders the TextInput component correctly', () => {
    const wrapper = shallow(<TextInput />);
    expect(wrapper).toMatchSnapshot();
  });
});
