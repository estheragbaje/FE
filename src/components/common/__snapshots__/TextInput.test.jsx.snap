// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders the TextInput component correctly 1`] = `ShallowWrapper {}`;
